src/canopy/llm/anyscale.py

```diff
@@ -41,23 +33,6 @@ def __init__(
         ae_base_url = base_url
         super().__init__(model_name, api_key=ae_api_key, base_url=ae_base_url, **kwargs)
 
-    @retry(
-        reraise=True,
-        stop=stop_after_attempt(3),
-        retry=retry_if_exception_type(
-            (json.decoder.JSONDecodeError, jsonschema.ValidationError)
-        ),
-    )
-    def enforced_function_call(
```

Comment:
I think it would be wise to wrap the parent's class function with a  `try: except:`.  
Then in case where user configured the wrong model (e.g. llama2) - we would catch the error coming from Anyscale, and re-raise with a clear actionable error message like `f"Model {self.model_name} doesn't support function calling. To use function calling capability, please select a different `model_name` in configuration.`

Reply:
Good suggestion. Added model check for function calls